lib/adapters/xhr.js

```diff
@@ -63,7 +55,30 @@ module.exports = function xhrAdapter(config) {
 
       // Clean up request
       request = null;
-    };
+    }
+
+    if ('onloadend' in request) {
+      // Use onloadend if available
+      request.onloadend = onloadend;
+    } else {
+      // Listen for ready state to emulate onloadend
+      request.onreadystatechange = function handleLoad() {
+        if (!request || request.readyState !== 4) {
+          return;
+        }
+
+        // The request errored out and we didn't get a response, this will be
+        // handled by onerror instead
+        // With one exception: request that using file: protocol, most browsers
+        // will return status as 0 even though it's a successful request
+        if (request.status === 0 && !(request.responseURL && request.responseURL.indexOf('file:') === 0)) {
+          return;
+        }
+        // readystate handler is calling before onerror or ontimeout handlers,
+        // so we should call onloadend on the next 'tick'
+        setTimeout(onloadend);
```

Comment:
this `setTimeout` appears to be causing test failures for me when using sinon/nise FakeServer with FakeTimers in Jest. And possibly with `nock`.
replacing this line with `onloadend()` causes my tests to pass again.

Reply:
Same for me. Thx for pointing it out. :) 
Will this be fixed in a new release, because this bug appears since v0.21.2. Is it already planned? :)

Reply:
For one project we ended up putting `flushPromises` and `await sandbox.clock.tickAsync()` / `sandbox.clock.tick` in the appropriate places in our custom wrappers for sinon FakeServer and requests to workaround this change so that we could use the latest version of axios in the project.

Reply:
Thanks for the response :) ... the `await clock.tickAsync();` worked for me. Thanks a lot!